src/odemis/gui/cont/acquisition.py

```diff
@@ -761,34 +733,39 @@ def _update_acquisition_time(self):
         self._panel.txt_cryosecom_est_time.SetLabel(txt)
 
     @call_in_wx_main
-    def _on_name_and_wavelength(self, VA_value):
+    def _on_stream_wavelength(self, _):
+        """
+        a VA callback that is called when the user changes
+         the emission or excitation wavelength
+        """
+        counts = self._panel.streams_chk_list.GetCount()
+        streams = [
+            self._panel.streams_chk_list.GetClientData(i) for i in range(counts)
+        ]
+        for i, s in zip(range(counts), acqmng.sortStreams(streams)):
+            # replace the unsorted streams with the sorted streams one by one
+            self._panel.streams_chk_list.Delete(i)
+            self._panel.streams_chk_list.Insert(s.name.value, i, s)
+            self._panel.streams_chk_list.Check(i)
+
+    @call_in_wx_main
+    def _on_stream_name(self, stream_name):
         """
         a VA callback that is called when the user changes
-         the name of a stream or the emission/excitation wavelength
-        """
-        if isinstance(VA_value, str):  # the user changed the stream name
-            counts = self._panel.streams_chk_list.GetCount()
-            chcklist_names = [
-                self._panel.streams_chk_list.GetString(i) for i in range(counts)
-            ]
-            streams = [
-                self._panel.streams_chk_list.GetClientData(i) for i in range(counts)
-            ]
-            stream_names = [s.name.value for s in streams]
-            for name, i, s in zip(chcklist_names, range(counts), streams):
-                if name not in stream_names:
-                    self._panel.streams_chk_list.Delete(i)
-                    self._panel.streams_chk_list.Insert(VA_value, i, s)
-                    self._panel.streams_chk_list.Check(i)
-        else:  # the user changed the wavelength
-            counts = self._panel.streams_chk_list.GetCount()
-            streams = [
-                self._panel.streams_chk_list.GetClientData(i) for i in range(counts)
-            ]
-            for i, s in zip(range(counts), acqmng.sortStreams(streams)):
-                # replace the unsorted streams with the sorted streams one by one
+         the name of a stream 
+        """
+        counts = self._panel.streams_chk_list.GetCount()
+        chcklist_names = [
+            self._panel.streams_chk_list.GetString(i) for i in range(counts)
+        ]
+        streams = [
+            self._panel.streams_chk_list.GetClientData(i) for i in range(counts)
+        ]
+        stream_names = [s.name.value for s in streams]
+        for name, i, s in zip(chcklist_names, range(counts), streams):
+            if name not in stream_names:
                 self._panel.streams_chk_list.Delete(i)
-                self._panel.streams_chk_list.Insert(s.name.value, i, s)
+                self._panel.streams_chk_list.Insert(stream_name, i, s)
                 self._panel.streams_chk_list.Check(i)
```

Comment:
I think you can make it a lot clearer by just iterating over all the streams once, and checking one-by-one that it's still the right name. Also, I think it's possible to use SetString() to just change the name, instead of deleting+inserting.
```python
counts = self._panel.streams_chk_list.GetCount()
for i in range(counts):
    sname = self._panel.streams_chk_list.GetString(i)
    s = self._panel.streams_chk_list.GetClientData(i)
    if s.name.value != sname:
           self._panel.streams_chk_list.SetString(i, s.name.value)
```

Reply:
indeed simpler and more compact